### ISSUE TITLE
Snapshot wait_for_completion only works as n URL parameter

### ISSUE THREAD

**Issue by kadarmarton** [elastic/elasticsearch#75284]
**Elasticsearch version** : 7.12.1

**Plugins installed**: []

**JVM version** :
openjdk version "16" 2021-03-16
OpenJDK Runtime Environment AdoptOpenJDK (build 16+36)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 16+36, mixed mode, sharing)

**OS version** : Linux 3.10.0-1160.el7.x86_64

**Description of the problem including expected versus actual behavior**:
The documentation of the [Create snapshot API](https://www.elastic.co/guide/en/elasticsearch/reference/current/create-snapshot-api.html#create-snapshot-api-request-body) lists the `wait_for_completion` parameter under the Request body section.
But in fact it only works as n URL parameter.

**Steps to reproduce**:

 1. Set up a cluster with a snapshot repository named `myrepo` (this might include setting up a shared file service, mounting it, setting the path.repo parameter accordingly on each node, restarting the cluster, and creating a repo on the mounted path)
 2. The following steps are presented as Dev Tools Console format requests.
 3. Create test data
```
PUT wait_for_completion_test_index/_doc/1
{ "testfield": "testdata" }
```
 4. Create snapshot with parameter embedded in request body:
```
POST _snapshot/myrepo/wait_for_completion_test_snapshot_1
{
	"indices": "wait_for_completion_test_index",
	"ignore_unavailable": false,
	"include_global_state": false,
	"wait_for_completion": true
}
```
But it returns immediately:
```
{
  "accepted" : true
}
```
 5. Create snapshot with parameter embedded in URL:
```
POST _snapshot/myrepo/wait_for_completion_test_snapshot_2?wait_for_completion=true
{
	"indices": "wait_for_completion_test_index",
	"ignore_unavailable": false,
	"include_global_state": false
}
```
This in fact returns synchronously:
```
{
  "snapshot" : {
    "snapshot" : "wait_for_completion_test_snapshot_2",
    "uuid" : "BkTqFLC5SGudlRjhyjUXlg",
    "version_id" : 7120199,
    "version" : "7.12.1",
    "indices" : [
      "wait_for_completion_test_index"
    ],
    "data_streams" : [ ],
    "include_global_state" : false,
    "state" : "SUCCESS",
    "start_time" : "2021-07-13T10:07:40.893Z",
    "start_time_in_millis" : 1626170860893,
    "end_time" : "2021-07-13T10:07:41.093Z",
    "end_time_in_millis" : 1626170861093,
    "duration_in_millis" : 200,
    "failures" : [ ],
    "shards" : {
      "total" : 1,
      "failed" : 0,
      "successful" : 1
    },
    "feature_states" : [ ]
  }
}
```
 6. Clean up:
```
DELETE _snapshot/myrepo/wait_for_completion_test_snapshot_1

DELETE _snapshot/myrepo/wait_for_completion_test_snapshot_2

DELETE wait_for_completion_test_index
```

**Comment by gaobinlong** [comment]
Maybe it's just a document issue, `wait_for_completion ` should be under the `Query parameters` section, which is same with the [Restore snapshot API](https://www.elastic.co/guide/en/elasticsearch/reference/current/restore-snapshot-api.html)

**Comment by jrodewig** [comment]
Thanks for raising this @kadarmarton. As @gaobinlong pointed out, this was a documentation bug. `wait_for_completion` is a query parameter, not a request body parameter.

We've fixed the bug with #76436 so I'm going to close this issue. Please feel free to re-open if I've overlooked anything.
